### ISSUE TITLE
Picroft - has library error on playing sound

### ISSUE THREAD

**Issue by robboz4** [MycroftAI/mimic#129]
I'm trying to get Mycroft to run on a Raspberry P3. I followed the instructions but get the error listed below when Mycroft is trying to speak. The microphone and speaker work, as it detects the wake word and chirps the speaker ready for input. I've also played sounds out of the speaker as a test. No-one has  pasted a reply to my request on the mycroft forum, so I thought I'd ask here.

```
67 - mycroft.client.speech.mic - DEBUG - Waiting for wake word...
2017-07-04 09:43:01,496 - requests.packages.urllib3.connectionpool - DEBUG - https://api.mycroft.ai:443 "POST /v1/stt?lang=en-US&limit=1 HTTP/1.1" 200 None
2017-07-04 09:43:02,386 - mycroft.client.speech.listener - DEBUG - STT: what time is it please
2017-07-04 09:43:02,470 - SpeechClient - INFO - Utterance: [u'what time is it please']
839253, max remaining 16
    ignoring unknown attr section 

sr/local/bin/mimic: error while loading shared libraries: libttsmimic_lang_vid_gb_ap.so.0: cannot open shared object file: No such file or directory
2017-07-04 09:43:02,738 - SpeechClient - INFO - Speak: Currently 09:43, AM
    ignoring unknown attr section aeacC
    invalid section len 51839253, max remaining 16
    ignoring unknown attr section 

sr/local/bin/mimic: error while loading shared libraries: libttsmimic_lang_vid_gb_ap.so.0: cannot open shared object file: No such file or directory
2017-07-04 09:43:02,860 - SpeechClient - INFO - Speak: 09:43, AM
    ignoring unknown attr section aeacC
    invalid section len 51839253, max remaining 16
    ignoring unknown attr section 

sr/local/bin/mimic: error while loading shared libraries: libttsmimic_lang_vid_gb_ap.so.0: cannot open shared object file: No such file or directory
p.so.0: cannot open shared object file: No such file or directory

```

There is also another person experiencing the same issue and they have had no response to their request either. 
So if anyone can assist that would be excellent. Let me know what kind of information you might need to further debug this issue.

Thanks in advanced.

**Comment by zeehio** [comment]
I don't know much about picroft... but it seems the file you are looking for should be installed by the mimic deb package at /usr/local/lib/libttsmimic_lang_vid_gb_ap.so.0 

Could you please answer these questions:
1. Does this file exist? `/usr/local/lib/libttsmimic_lang_vid_gb_ap.so.0`
2. Does `ldconfig -v 2>/dev/null | grep -v ^$'\t'` print `/usr/local/lib` among other directories?
3. What does `apt show mimic` print?

Thanks!

**Comment by robboz4** [comment]
Here are the answers:
1) ls -l /usr/local/lib/libttsmimic_lang_vid_gb_ap.so.0
lrwxrwxrwx 1 mycroft mycroft 35 Mar 15 17:05 /usr/local/lib/libttsmimic_lang_vid_gb_ap.so.0 -> libttsmimic_lang_vid_gb_ap.so.0.0.0

2) ldconfig -v 2>/dev/null | grep -v ^$'\t'
/opt/vc/lib:
/lib/arm-linux-gnueabihf:
/usr/lib/arm-linux-gnueabihf:
/usr/lib/arm-linux-gnueabihf/libfakeroot:
/usr/local/lib:
/lib:
/usr/lib:
/usr/lib/arm-linux-gnueabihf/vfp: (hwcap: 0x0000000000000040)
/usr/lib/arm-linux-gnueabihf/neon: (hwcap: 0x0000000000001000)
/usr/lib/arm-linux-gnueabihf/vfp/neon: (hwcap: 0x0000000000001040)
/usr/lib/arm-linux-gnueabihf/neon/vfp: (hwcap: 0x0000000000001040)

3) sudo apt show mimic
Package: mimic
Version: 1.2.0.2
Maintainer: Arron Atchison <arron.atchison@mycroft.ai>
Priority: optional
Section: base
Download-Size: 84.4 MB
Installed-Size: unknown
APT-Manual-Installed: no
APT-Sources: http://repo.mycroft.ai/repos/apt/debian/ debian/main armhf Packages
Description: mimic

Cheers
Dave

**Comment by el-tocino** [comment]
Which version are you on? Run "dpkg -l| grep mycroft"
Anything else running on your pi?
Can you try running: sha1sum /usr/local/lib/libttsmimic_lang_vid_gb_ap.so.0

**Comment by robboz4** [comment]
There should only be Mycroft running as I used the "picroft" image they recommended. Here are your answers:
1) dpkg -l| grep mycroft
ii  mycroft-core                       0.8.17                          armhf        mycroft-core

2) pi@picroft:~ $  sha1sum /usr/local/lib/libttsmimic_lang_vid_gb_ap.so.0
e23cee4eaf8bc61d69b47f969ec463e04c56a385  /usr/local/lib/libttsmimic_lang_vid_gb_ap.so.0

Thanks

**Comment by el-tocino** [comment]
Can you, as a test, try setting a different voice? (at home.mycroft.ai, then settings, then advanced)

**Comment by robboz4** [comment]
I changed it to another and the same issue. I also spotted this but I don't think it's the issue:

2017-07-05 21:48:58,281 - mycroft.configuration - DEBUG - Configuration '/home/mycroft/.mycroft/mycroft.conf' loaded
 But I could be wrong. I read that conf file is read from somewhere else and a local file would override it. But I could be wrong...
I did reboot after changing the voice as a further test.

**Comment by zeehio** [comment]
Still trying to understand what is going on. Could you please: `dpkg --verify mimic` or even trying to reinstall mimic with: `apt reinstall mimic`?

**Comment by robboz4** [comment]
1) dpkg --verify mimic
??5??????   /usr/local/lib/libttsmimic_lang_vid_gb_ap.so.0.0.0

2) I did a sudo apt-get remove mimic, followed by 
Sudo apt-get install mimic
As reinstall was not an option on my system. I then did a reboot.
Now mycroft is not even triggering when the wake word is used. I tried an update to see if that would fix, but no. So now nothing works... 
it looks like Mycroft  on a Raspberry Pi doesn't seem to work. I have no idea how the folks got it work as they stated. I followed all they guides on Picroft, but now it's not even triggering.
Thanks for trying.

**Comment by el-tocino** [comment]
Your post reports a failed hash on the mimic file.  
https://askubuntu.com/questions/792553/dpkg-v-what-does-the-output-mean

I think you might want to re-image your SD card and try again.

**Comment by robboz4** [comment]
Ok will do! It will be a couple of days before I get to it, but will post a response.

Thanks

**Comment by robboz4** [comment]
I loaded a new SD card today and here are the results:

1) New image booted and issued the pairing key via the speaker. During the boot it said it  couldn't verify mimic my-croft-core
2) Added a new device to Mycroft.ai and the two paired.
3) Mycroft than responded to the wake phrase and replied with audio! Success, but wait...
4) Configured the Pi for my location (California US), expanded file system, added wait on network for boot, changed the password and the Pi rebooted.
5) During reboot it updated mimic. Then tried to update mycroft, but I hit the key issue.
6) Added the keys and rebooted
7) Everything updated Mycroft chirps on wake phases. Prints out no error but also doesn't speak.
8) I did the dpkg --verify mimic command and it returned nothing. 
9) did the dpkg command on mycroft as listed above and got the same value.
10) Here are the errors in the mycroft logs:

/var/log $ grep error my*.log
mycroft-skills.log:2017-07-07 11:03:31,180 - mycroft.messagebus.client.ws - ERROR - Exception("Uncaught 'error' event.",)
mycroft-speech-client.log:ALSA lib conf.c:4259:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
mycroft-speech-client.log:ALSA lib conf.c:4738:(snd_config_expand) Evaluate error: No such file or directory
mycroft-speech-client.log:ALSA lib conf.c:4259:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
mycroft-speech-client.log:ALSA lib conf.c:4738:(snd_config_expand) Evaluate error: No such file or directory
mycroft-speech-client.log:ALSA lib conf.c:4259:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
mycroft-speech-client.log:ALSA lib conf.c:4738:(snd_config_expand) Evaluate error: No such file or directory
mycroft-speech-client.log:ALSA lib conf.c:4259:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
mycroft-speech-client.log:ALSA lib conf.c:4738:(snd_config_expand) Evaluate error: No such file or directory
mycroft-speech-client.log:ALSA lib conf.c:4259:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
mycroft-speech-client.log:ALSA lib conf.c:4738:(snd_config_expand) Evaluate error: No such file or directory
mycroft-speech-client.log:ALSA lib conf.c:4259:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
mycroft-speech-client.log:ALSA lib conf.c:4738:(snd_config_expand) Evaluate error: No such file or directory
mycroft-speech-client.log:ALSA lib conf.c:4259:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
mycroft-speech-client.log:ALSA lib conf.c:4738:(snd_config_expand) Evaluate error: No such file or directory
mycroft-speech-client.log:ALSA lib conf.c:4259:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
mycroft-speech-client.log:ALSA lib conf.c:4738:(snd_config_expand) Evaluate error: No such file or directory
mycroft-speech-client.log:ALSA lib conf.c:4259:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
mycroft-speech-client.log:ALSA lib conf.c:4738:(snd_config_expand) Evaluate error: No such file or directory
mycroft-speech-client.log:ALSA lib conf.c:4259:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
mycroft-speech-client.log:ALSA lib conf.c:4738:(snd_config_expand) Evaluate error: No such file or directory
mycroft-speech-client.log:ALSA lib conf.c:4259:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
mycroft-speech-client.log:ALSA lib conf.c:4738:(snd_config_expand) Evaluate error: No such file or directory
mycroft-wifi-setup-client.log:2017-07-07 11:03:30,965 - mycroft.messagebus.client.ws - ERROR - Exception("Uncaught 'error' event.",)

So to summarize, it worked as is from the original SD image, but updates to either mimic or mycroft seem to break it. Or setting up a US based localization... 

What's next?

Cheers

Dave

**Comment by el-tocino** [comment]
There's a new release from this weekend. I'd try that first, it has a lot of fixes in it.  If that's still a problem...well..will look further.

**Comment by robboz4** [comment]
A quick update> I find up the Pi today, not making any changes and Mycroft now responds!!! However, It will only respond to Hey Mycroft: What time is it?

The Eiffel tower one doesn't reply or the question on how are you.

But it seem like I can now do more with it.

Cheers

Dave

**Comment by robboz4** [comment]
sorry, the posts crossed. I will try an update to mimic. During the boot nothing was updated according to  the log.

**Comment by aatchison** [comment]
Hmm, that's very strange, and I've never seen logs like that before. It looks like an audio subsystem problem though. We are updating the core again very soon here (to 0.8.18,) but we haven't changed the mimic package recently.

**Comment by robboz4** [comment]
I've spotted a couple of issues that should be reported to different parts of the project, but will list them here for notes on my ongoing pi installation:
1) Mycroft repeated the phase - that is two voice outputs for one request:

2017-07-10 14:24:59,928 - mycroft.client.speech.mic - DEBUG - Waiting for wake word...
2017-07-10 14:24:59,951 - requests.packages.urllib3.connectionpool - DEBUG - https://api.mycroft.ai:443 "POST /v1/stt?lang=en-US&limit=1 HTTP/1.1" 200 None
2017-07-10 14:25:00,834 - mycroft.client.speech.listener - DEBUG - STT: alarm list
2017-07-10 14:25:00,835 - SpeechClient - INFO - Utterance: [u'alarm list']
2017-07-10 14:25:00,856 - SpeechClient - INFO - Speak: There is an alarm set for 0 minutes and 42 seconds from now
e20.wav' : Signed 16 bit Little Endian, Rate 44100 Hz, Mono
4144e4e20.wav' : Signed 16 bit Little Endian, Rate 44100 Hz, Mono
 alarm set for 0 minutes and 41 seconds from now
peak: There is an alarm set for 0 minutes and 41 seconds from now

2) I added  timer for 60 minutes. It counted down but never activated instead I got this response to a timer list:
2017-07-10 14:26:12,357 - mycroft.client.speech.mic - DEBUG - Waiting for wake word...
2017-07-10 14:26:13,068 - mycroft.client.speech.listener - DEBUG - STT: alarm list
2017-07-10 14:26:13,068 - SpeechClient - INFO - Utterance: [u'alarm list']
2017-07-10 14:26:13,097 - SpeechClient - INFO - Speak: There is an alarm set for -1 hours and 59 minutes from now
Playing WAVE '/tmp/mycroft/cache/tts/dbdae630bf1d0373116e75647b355052.wav' : Signed 16 bit Little Endian, Rate 44100 Hz, Mono
2017-07-10 14:26:20,567 - SpeechClient - INFO - Speak: You have an alarm on -1 hours and 59 minutes from now

Probably a math error in the alarm time code.

Anyway. Thanks for the help getting it going. I'll test some more and post any issues to correct threads.

**Comment by el-tocino** [comment]
These should both go over in the mycroft-core repo. There's a duplicate voice one already that may be of interest.  https://github.com/MycroftAI/mycroft-core/issues/597 Don't see anything on the alarm one.  Might want to open a new issue and get logs from both setting and checking it if you can.

**Comment by robboz4** [comment]
We can close this now. With the 08.18 update there are no  errors and no repeats.  I still have the pending alarm issue to report. Thanks again for the help getting my Picroft running.

**Comment by oren** [comment]
I think i might have similar issue - https://community.mycroft.ai/t/mycroft-doesnt-seems-to-hear-me/1878/2
